src/lib/components/explorer/export/ExportStepper.svelte

```diff
@@ -0,0 +1,285 @@
+  import type { ExportRowInterface } from '$lib/models/ExportRow';
+  import type { QueryRequestInterface } from '$lib/models/api/Request';
+  import { ProgressRadial } from '@skeletonlabs/skeleton';
+  import ErrorAlert from '$lib/components/ErrorAlert.svelte';
+  import ExportStore from '$lib/stores/Export';
+  let { exports } = ExportStore;
+  import { state } from '$lib/stores/Stepper';
+  import { goto } from '$app/navigation';
+  import { type DataSet, type DatasetError } from '$lib/models/Dataset';
+  import { createDatasetName } from '$lib/services/datasets';
+
+  export let query: QueryRequestInterface;
+  export let showTreeStep = false;
+  export let rows: ExportRowInterface[] = [];
+  let statusPromise: Promise<string>;
+  let preparePromise: Promise<void>;
+  let datasetNameInput: string = '';
+  let picsureResultId: string = '';
+  let lockDownload = true;
+  let error: string = '';
+  let namedDataset: DataSet;
+  $: datasetId = '';
+  $: canDownload = true;
+  $: apiExport = false;
+  const columns = [
+    { dataElement: 'variableName', label: 'Variable Name', sort: true },
+    { dataElement: 'description', label: 'Variable Description', sort: true },
+    { dataElement: 'type', label: 'Type', sort: true },
+  ];
+
+  async function onCompleteHandler(): Promise<void> {
+    try {
+      const res = await api.post(`picsure/query/${datasetId}/result`, {});
+      const responseDataUrl = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
+      if (browser) {
+        const link = document.createElement('a');
+        link.href = responseDataUrl;
+        link.download = 'pic-sure-data.csv';
+        document.body.appendChild(link);
+        link.click();
+        document.body.removeChild(link);
+      }
+      if (!apiExport) {
+        $state.current = 0;
+        goto('/dataset/' + namedDataset.uuid);
+      }
+    } catch (error) {
+      console.error('Error in onCompleteHandler', error);
+    }
+  }
+  function onNextHandler(e: CustomEvent): void {
+    console.log('event:next', e);
+    if (e.detail.step === 0 && !showTreeStep) {
+      preparePromise = handleFirstStep();
+    }
+    if (e.detail.step === 1 && showTreeStep) {
+      console.log('event:next', e);
+      //TODO: Load tree
+    } else if ((e.detail.step === 1 && !showTreeStep) || (showTreeStep && e.detail.step === 2)) {
+      createNamedDataset();
+    }
+    if (e.detail.state.total - 1 === e.detail.step + 1) {
+      statusPromise = new Promise((resolve, reject) => {
+        const interval = setInterval(async () => {
+          const status = await checkExportStatus(picsureResultId);
+          if (status !== 'PENDING' || status !== 'RUNNING' || status !== 'QUEUED') {
+            clearInterval(interval);
+            if (status === 'ERROR') {
+              lockDownload = true;
+              reject(status);
+            } else {
+              lockDownload = false;
+              resolve(status);
+            }
+          }
```

Comment:
```suggestion
         if (status === 'ERROR') {
            lockDownload = true;
            clearInterval(interval);
            reject(status);
          } else if (!['PENDING', 'RUNNING', 'QUEUED'].includes(status)) {
            lockDownload = false;
            clearInterval(interval);
            resolve(status);
          }
```
This looks a bit clearer to me, even if it's slightly redundant in one place.